react-app/src/components/Opensearch/main/Filtering/Export/index.tsx

```diff
@@ -58,37 +59,63 @@ export const OsExportData: FC<{
     });
 
     csvExporter.generateCsv(exportData);
-
     setLoading(false);
   };
 
+  const handleExport = async () => {
+    if (disabled) {
+      return;
+    }
+
+    if (count > EXPORT_LIMIT) {
+      setShowAlert(true);
+      return;
+    }
+
+    await exportToCsv();
+  };
+
   return (
-    <TooltipProvider>
-      <Tooltip disableHoverableContent={true}>
-        <TooltipTrigger asChild className="disabled:pointer-events-auto">
-          <Button
-            variant="outline"
-            onClick={handleExport}
-            disabled={loading || disabled}
-            className="w-full xs:w-fit hover:bg-transparent self-center h-10 flex gap-2"
-            data-testid="tooltip-trigger"
-          >
-            {loading && (
-              <motion.div
-                animate={{ rotate: "360deg" }}
-                transition={{ repeat: Infinity, duration: 0.5 }}
-              >
-                <Loader className="w-4 h-4" />
-              </motion.div>
-            )}
-            {!loading && <Download className="w-4 h-4" />}
-            <span className="prose-sm">Export</span>
-          </Button>
-        </TooltipTrigger>
-        {disabled && (
-          <TooltipContent data-testid="tooltip-content">No records available</TooltipContent>
-        )}
-      </Tooltip>
-    </TooltipProvider>
+    <>
+      <ConfirmationDialog
+        open={showAlert}
+        title={"Export limit reached"}
+        body={"Only the first 10,000 records can be exported. Try filtering to get fewer results."}
+        acceptButtonText={"Export"}
+        aria-labelledby={"Export limit confirmation dialog."}
```

Comment:
No need for curly braces
```suggestion
        title="Export limit reached"
        body="Only the first 10,000 records can be exported. Try filtering to get fewer results."
        acceptButtonText="Export"
        aria-labelledby="Export limit confirmation dialog."
```

Reply:
Updated. Maybe a good lint rule 🤔